includes/reader-revenue/woocommerce/class-woocommerce-connection.php

```diff
@@ -368,36 +376,34 @@ public static function get_contact_order_metadata( $order, $payment_page_url = f
 	/**
 	 * Get data for a customer to sync to the connected ESP.
 	 *
-	 * @param \WC_Customer    $customer Customer object.
-	 * @param \WC_Order|false $order Order object to sync with. If not given, the last successful order will be used.
-	 * @param bool|string     $payment_page_url Payment page URL. If not provided, checkout URL will be used.
-	 * @param bool            $is_new Whether the order is new and should count as the last payment amount.
+	 * @param int|\WC_Customer $customer Customer object or customer ID.
+	 * @param \WC_Order|false  $order Order object to sync with. If not given, the last successful order will be used.
+	 * @param bool|string      $payment_page_url Payment page URL. If not provided, checkout URL will be used.
 	 *
 	 * @return array|false Contact data or false.
 	 */
-	public static function get_contact_from_customer( $customer, $order = false, $payment_page_url = false, $is_new = false ) {
+	public static function get_contact_from_customer( $customer, $order = false, $payment_page_url = false ) {
 		if ( ! is_a( $customer, 'WC_Customer' ) ) {
-			return false;
+			$customer = new \WC_Customer( $customer );
 		}
 
-		$metadata       = [];
-		$order_metadata = [];
-		$last_order     = self::get_last_successful_order( $customer );
+		$metadata = [];
 
 		$metadata[ Newspack_Newsletters::get_metadata_key( 'account' ) ]           = $customer->get_id();
 		$metadata[ Newspack_Newsletters::get_metadata_key( 'registration_date' ) ] = $customer->get_date_created()->date( Newspack_Newsletters::METADATA_DATE_FORMAT );
 		$metadata[ Newspack_Newsletters::get_metadata_key( 'total_paid' ) ]        = \wc_format_localized_price( $customer->get_total_spent() );
 
-		// If a more recent order exists, use it to sync.
-		if ( ! $order || ( $last_order && $order->get_id() !== $last_order->get_id() ) ) {
-			$order = $last_order;
+		// If no order is passed as the argument, use the last successful order.
+		if ( $order === false ) {
+			$order = self::get_last_successful_order( $customer );
 		}
 
 		// Get the order metadata.
+		$order_metadata = [];
 		if ( $order ) {
-			$order_metadata = self::get_contact_order_metadata( $order, $payment_page_url, $is_new );
+			$order_metadata = self::get_contact_order_metadata( $order, $payment_page_url );
 		} else {
-			// If the customer has no successful orders, clear out subscription-related fields.
+			// If the customer has no successful orders or if no order is provided???, clear out subscription-related fields.
```

Comment:
That seems like un unreasonable amount of question marks 😄

Reply:
That was more of a note to self :D Fixed in 93428490db4ebdbaaff954524cc71ae1d4d3fe39